Automatisk sjekk av kodestil ved pull request

haugedal opened the pull request.

haugedal committed:
Automatisk sjekk av kodestil ved pull request

statkarl commented on .github/workflows/style.yaml:
```diff
@@ -0,0 +1,74 @@
+        shell: Rscript {0}
+
+      - name: Verify Changed files
+        uses: tj-actions/verify-changed-files@v16
+        id: verify-changed-files
+
+      - name: Run step only when any of the above files change.
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        run: |
+          echo "::error::Filer med feil formatering: ${{ steps.verify-changed-files.outputs.changed_files }}"
+          exit 1
```
Kan kanskje alt erstattast med følgjande `run`-steg:

    git diff --exit-code

Kommandoen vil gje exit-verdi lik 1 viss nokon filer er endra. Og utdata vil jo vera resultatet av diff-kommandoen, som vil visa både kva filer er feilformaterte og korleis dei er feilformaterte. Og det er jo nettopp den informasjon me treng. Verdt eit forsøk? Og så slepp me å vera avhengig av kode i eit depot me ikkje har kontroll over.

statkarl replied:
Kanskje òg med `--ws-error-highlight=all`, for å visa kor det er endringar i blankteikn.

haugedal replied:
Skal testa. 👍

statkarl replied:
Ser ut til å funka. Prøv no med `--patch-with-stat --color=always` i tillegg.

statkarl replied:
Lekkert. Effekten av `--ws-error-highlight=all` vert ikkje synleg, men det er vel GitHub sin feil. Bør bli ståande, så kanskje vert det fiksa ein gong i framtida.

haugedal committed:
Ikkje style mappe med testfiler for rapwhale-style

Dette er filer som skal innehalda kode med feil kodestil
for å testa at rapwhale_style() funkar.

statkarl commented on .github/workflows/style.yaml:
```diff
@@ -0,0 +1,67 @@
+          extra-packages: any::styler
+          needs: styler
+
+      - name: Document
+        run: roxygen2::roxygenise()
+        shell: Rscript {0}
```
Treng me denne? Og han vil vel laga nye filer, noko me ikkje vil ha.

haugedal replied:
Nei, det går kanskje greitt utan.

haugedal replied:
Får ikkje installert `rapwhale` utan dette steget:

```
ERROR: a 'NAMESPACE' file is required
```

statkarl replied:
Ja, det var det, ja. Får vel stå, då.

statkarl commented on R/aggreger_ki_prop.R:
```diff
@@ -148,3 +148,5 @@ aggreger_ki_prop = function(d_ki_ind, alfa = 0.05) {
 
   d_sammendrag
 }
+
+#Kommentar som ikkje fylgjer kodestilen
```
Desse må vekk.

haugedal replied:
Ja, dei er berre med for å få testa den automatiske sjekken.

statkarl replied:
Dei trengst vel ikkje for å testa det? Styler vert jo køyrt på alle filene, ikkje berre dei endra. Og me har mykje som ikkje følgjer styler-formateringa.

haugedal replied:
Ja, men sjølve GitHub-actionen vert berre køyrt viss filer med R-kode er endra:
```
on:
  pull_request:
    branches: [main]
    paths: ["**.[rR]", "**.[qrR]md", "**.[rR]markdown", "**.[rR]nw", "**.[rR]profile"]
```

statkarl replied:
Å! Okei. Men du må fjerna det viss eg skal godkjenna PR-en. Det er det einaste som manglar no.

haugedal replied:
Fjerna (og rebasert på main).

statkarl commented on .github/workflows/style.yaml:
```diff
@@ -0,0 +1,67 @@
+        run: styler::style_pkg(style = rapwhale::rapwhale_style, exclude_dirs = "tests/testthat/rapwhale-style")
+        shell: Rscript {0}
+
+      - name: Check for changed files
```
Dette vert overskrifta i loggen, så kanskje finna noko litt meir forståeleg, eks. `Filer som ikkje følgjer kodestilen vår`?

haugedal commented:
Skal kodestilsjekken vera obligatorisk, forresten?

statkarl commented:
> Skal kodestilsjekken vera obligatorisk, forresten?

Ikkje no, men etter at me har restyla heile pakken. Bør kanskje venta til @FatemehGhavidel har lagt inn endringane sine (om nytt røyr) først (før me restylar), for å unngå Git-konfliktar.

haugedal commented:
> > Skal kodestilsjekken vera obligatorisk, forresten?
> 
> Ikkje no, men etter at me har restyla heile pakken. Bør kanskje venta til @FatemehGhavidel har lagt inn endringane sine (om nytt røyr) først (før me restylar), for å unngå Git-konfliktar.

OK.
